WordPress/gutenberg
# Revert "Patterns: fix capabilities settings for pattern categories

This reverts commit 5b10d27ec406cbdec84e14408cd350ba9e8c544a.

## What?
Reverts the changes made in #55379 that prevents Author level users from adding pattern categories. 

## Why?
It was thought that Authors should be able to choose from existing categories but not add new ones, and this is probably still the idea, but there are potential issues with modifying the taxonomy capabilities to disallow this, [as mentioned here](https://github.com/WordPress/wordpress-develop/pull/5510#issuecomment-1772921357). The best option is probably to back this out and leave it with authors able to add new pattern categories at this point in the 6.4 release cycle.

## How?
A straight revert of 5b10d27ec406cbdec84e14408cd350ba9e8c544a.

## Testing Instructions
- Log in as a site admin or editor and add a new pattern and assign some new categories and make sure they save and can be viewed as expected in site editor patterns pages and in the post editor inserter patterns tab
- Don't test the Author user as they can't view the categories they add until the [change noted here](https://github.com/WordPress/gutenberg/pull/55532#issuecomment-1773894853) is reimplemented
